tools/chapel-py/chplcheck/rules.py

```diff
@@ -126,13 +126,19 @@ def recurse(node, skip_direct = False):
     def MisleadingIndentation(context, root):
         prev = None
         for child in root:
+            if isinstance(child, Comment): continue
             yield from MisleadingIndentation(context, child)
 
             if prev is not None:
                 if child.location().start()[1] == prev.location().start()[1]:
                     yield child
 
+            prev = None
             if isinstance(child, Loop) and child.block_style() == "implicit":
                 grandchildren = list(child)
-                if len(grandchildren) > 0:
-                    prev = list(grandchildren[-1])[0]
+                # safe to access [-1], implicit loops must have at least 1 child
```

Comment:
Any loop must have at least one child -- the block, be it `{}` or `do` :)

Reply:
I can compile `for i in 1..10 {}`?